lib/codeowners/cleaner.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'git'
+require 'logger'
+
+require_relative 'checker/code_owners'
+require_relative 'checker/file_as_array'
+require_relative 'checker/whitelist'
+
+module Codeowners
+  # Cleans a codeowner file by sorting, grouping and removing
+  # useless patterns, all automatically
+  class Cleaner
+    # Get repo metadata and compare with the owners
+    def initialize(repo)
+      @git = Git.open(repo, log: Logger.new(IO::NULL))
+      @repo_dir = repo
+    end
+
+    def clean!
```

Comment:
Hey, great work. That method reads really nicely 👏  
Is there a reason for the bang notation for the method name and for the methods inside? Do we have a specific convention in this codebase? Otherwise, bang is mostly used as a warning. Like in-place mutations or potential error behavior.

Reply:
Ah, sorry. I was following ruby's convention that ! represents in-place mutations (as in the Array methods).

In Elixir, I'm used to just get the result from one and pass to another, but in Ruby since it is OO, there is no pipeline operator (I think it was voted off in ruby 2.7).

The normal way to do is as you suggested down there, creating a class with the chaining pattern. (as it reads exactly like pipe in FP languages).

So I think I'll go with your suggested approach!